### ISSUE TITLE
Using aiida-fleur with develop versions of Fleur

### ISSUE THREAD

**Issue by janssenhenning** [JuDFTteam/aiida-fleur#128]
This issue concerns both masci-tools and aiida-fleur, but I think aiida-fleur is the right place for this, since the issue is more apparent here

The file version of the current develop version is (and should be) higher than the latest available version in masci-tools. The reason for this is the easier distinction between invalid input files for the 5.1 release and files from the develop version, which just contain switches and inputs that are not in the release Schema. However, this requires some thought about how to use aiida-fleur with the develop version of fleur. Since the changes are very minor at the moment (only a couple optional switches that are new) We could enable this with two changes

First the `FleurinpData`, validates the XML files too early and does not make use of the fallback mechanism in the inpxml_parser. This is easy too fix and should be done.

The other place the user has to adjust is when modifying the fleurinpData. The first time you modify a develop version fleurinpData, you will need to set `fleurInputVersion` to `0.34`

The bigger question is how this should be handled in general. At the moment this quick fix can be done and virtually everything still works. But when the schema has larger changes, there might be no way around forcing the user to insert the schema into their masci-tools installation. I'm strongly against adding the fleur schema to the masci-tools repo before it is fixed by a release, since this can only lead to incompatibilities and headaches down the road

**Comment by broeder-j** [comment]
The last release of masci-tools can only be compatible to the last release of fleur and maybe sometimes to develop. 
In my view it would be fine to have the develop fleur schema always in masci-tools develop directly. (one has to think what happens on releases if we do this, or if that would bind masci-tools releases to fleur to avoid incompatibilities). So if you like to use the fleur develop branch with aiida-fleur you should also be on masci-tools develop.

I also understand that you are against this, but for small changes the schema it should not be a problem or?
One question would be: What has to be done on the masci-tools and fleur side to avoid these incompatibilities, i.e in terms of tests? If fleur would increase the version everytime for every schema change, i.e if they are bundled always on a branch and then merged into develop with a version increase, would that solve the problem?

On the other side, these two fall backs you suggest for aiida-fleur I would do them anyway. And in the modification case we give a warning like 'I do not know this version yet, I try to fall back to latest version I know, which might lead to unexpected errors'.

**Comment by janssenhenning** [comment]
Okay I think I will start by putting in the two fallbacks I proposed, since Gregor wants to use the develop version of fleur with aiida-fleur 😄

**Comment by janssenhenning** [comment]
@broeder-j After thinking about it for some more time I think we can solve this only in one place. The function `load_inpxml` on FleurinpData can check if it's own input version is the same as in the loaded schema dictionary (This is how this case is also detected in the parsers) and can ignore the validation errors in this case.

I would like to still output the validation errors as warnings. Do you happen to know if `Data` objects in aiida also have configured loggers?

**Comment by broeder-j** [comment]
Also I would like to run a nightly build, either on the our CI, or on github against fleur develop, with the full workflow regression, tests, because currently they are never run. Back then I ran them locally on my machine.
We could also test all input files of fleur regression tests, if we can create input for them. (ideal if the calculation through aiida gives the same result as the tests, but that I do not know howto do and is maybe overkill).

Also at some point we want to have calculation importers.

@janssenhenning: to the logging: not as the processes due, but i think if you would log something from a data object it would go to the daemon/aiida log, (or some other default aiida log file) before one could watch it live via `verdi daemon logshow`. Overall, I am not sure without checking and my knowledge is not up to date.

**Comment by janssenhenning** [comment]
@broeder-j You are right there is a logger but it just relays the log messages to the notebook or the daemon depending, where it is run. It could be that after implementing it, there could be a lot of warnings thrown around. If there are too many warnings we can look into how to reduce the number of warnings

**Comment by janssenhenning** [comment]
@broeder-j Probably outputting the complete validation errors is overkill, we can tell users to check the parser_info property since the validation errors will be in there. So a simple warning should be enough
